core-common/src/main/java/org/glassfish/jersey/model/internal/CommonConfig.java

```diff
@@ -674,13 +690,17 @@ private void configureFeatures(InjectionManager injectionManager,
                 // init lazily
                 featureContextWrapper = new FeatureContextWrapper(this, injectionManager);
             }
-            final boolean success = feature.configure(featureContextWrapper);
 
-            if (success) {
-                processed.add(registration);
-                configureFeatures(injectionManager, processed, resetRegistrations(), managedObjectsFinalizer);
-                enabledFeatureClasses.add(registration.getFeatureClass());
-                enabledFeatures.add(feature);
+            final RuntimeType runtimeTypeConstraint = registration.getFeatureRuntimeType();
+            if (runtimeTypeConstraint == null || type.equals(runtimeTypeConstraint)) {
```

Comment:
This check should be performed before line 674, so we do not create an injected instance of a Feature that should not exist in this type of runtime.